src/types.rs

```diff
@@ -370,9 +399,333 @@ impl From<ReturnCode> for u8 {
     }
 }
 
+/// RPC payload
+pub type RpcPayload<'a> = &'a [u8];
+
+/// SD payload
+#[derive(Debug, PartialEq)]
+pub struct SdPayload {
+    /// Flags
+    pub flags: SdFlags,
+    /// Entries
+    pub entries: Vec<SdEntry>,
+    /// Options
+    pub options: Vec<SdOption>,
+}
+
+impl SdPayload {
+    /// Length
+    pub fn len(&self) -> usize {
```

Comment:
struct `SdPayload` has a public `len` method, but no `is_empty` method

note: `#[warn(clippy::len_without_is_empty)]` on by default

Reply:
renamed to length()